op-supervisor/supervisor/backend/backend_test.go

```diff
@@ -697,3 +697,59 @@ func TestFailsafeEnabledConfigInitialization(t *testing.T) {
 		})
 	}
 }
+
+func TestFailsafeOnInvalidation(t *testing.T) {
+	logger := testlog.Logger(t, log.LvlInfo)
+	m := metrics.NoopMetrics
+	dataDir := t.TempDir()
+	fullCfgSet := fullConfigSet(t, 1)
+
+	testCases := []struct {
+		name                   string
+		failsafeOnInvalidation bool
+		expectFailsafeEnabled  bool
+	}{
+		{
+			name:                   "FailsafeOnInvalidationEnabled",
+			failsafeOnInvalidation: true,
+		},
+		{
+			name:                   "FailsafeOnInvalidationDisabled",
+			failsafeOnInvalidation: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			cfg := &config.Config{
+				Version:                "test",
+				FullConfigSetSource:    fullCfgSet,
+				SynchronousProcessors:  true,
+				MockRun:                false,
+				SyncSources:            &syncnode.CLISyncNodes{},
+				Datadir:                dataDir,
+				FailsafeEnabled:        false, // Start with failsafe disabled
+				FailsafeOnInvalidation: tc.failsafeOnInvalidation,
+			}
+
+			ex := event.NewGlobalSynchronous(context.Background())
+			b, err := NewSupervisorBackend(context.Background(), logger, m, cfg, ex)
+			require.NoError(t, err)
+
+			// Verify that failsafe starts disabled
+			enabled, err := b.GetFailsafeEnabled(context.Background())
+			require.NoError(t, err)
+			require.False(t, enabled, "failsafe should start disabled")
+
+			// Emit InvalidateLocalSafeEvent
+			b.OnEvent(context.Background(), superevents.InvalidateLocalSafeEvent{
```

Comment:
Nit: Would it be better to call `b.emitter.Emit` here? It would integrate a little more of the events system into the test which I think would be an improvement.

Reply:
It would test that the event system faithfully calls `OnEvent` for all derivers, which is covered in every e2e/integration test, as well as tests for the event system. It would indeed be more accurate, but not in service of this test.